_source/_posts/2022-01-13-node-deploy-to-heroku.md

```diff
@@ -0,0 +1,492 @@
+---
+layout: blog_post
+title: ""
+author:
+by: advocate|contractor
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness|conversion
+---
+
+## Introduction
+
+Heroku is a platform as a service that supports many languages.  Initially, it was developed to support Ruby sites but now supports an array of languages including javascript with Node.js.  Heroku also has docker support so you can deploy just about anything to it.  
+
+However, in this tutorial, you will learn how to build a small application using the `express` framework for Node.js.  You will then secure that application using Okta by creating an application in Okta for web applications and integrating the Okta OIDC middleware with your application.  Finally, you will learn how to deploy the application to Heroku, update your Okta application for your Heroku site, and how to set the environment variables for your app in Heroku.
+
+To continue you will need:
+
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [Node.Js](https://nodejs.org/en/)
+- An [Okta Developer Account](https://developer.okta.com/) (free forever, to handle your OAuth needs)
+- A [Heroku Account](https://signup.heroku.com/)
+- A [Git Repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)
```

Comment:
Does it make sense to list Git as a prerequisite? They can, later on, create a repo with `git init`.

Reply:
I'd leave it.  Otherwise we're just splitting hairs about "what is a prereq?"  Is an Okta Account a prereq because they can just do that from the the CLI?  The article there is something Heather Downing adding a little bit ago, and if its good enough for her its good enough for me.